packages/icons-workflow/bin/build.js

```diff
@@ -82,6 +84,40 @@ let manifestListings = `\r\nexport const iconManifest = [\r\n`;
 
 const defaultIconImport = `import { DefaultIcon as AlternateIcon } from '../DefaultIcon.js';\r\n`;
 
+// A function that process the raw svg name and returns the component name
+export const getComponentName = (i) => {
+    let id = path
+        .basename(i, '.svg')
+        .replace('S2_Icon_', '')
+        .replace('S_', '')
+        .replace('_20_N', '')
+        .replace('_22x20_N', '')
+        .replace('_18_N@2x', '');
+
+    if (id.search(/^Ad[A-Z]/) !== -1) {
+        id = id.replace(/^Ad/, '');
+        id += 'Advert';
+    }
+
+    if (id === 'UnLink') {
+        id = 'Unlink';
+    }
+    if (id === 'TextStrikeThrough') {
+        id = 'TextStrikethrough';
+    }
+
+    let ComponentName = id === 'github' ? 'GitHub' : Case.pascal(id);
+
+    if (ComponentName === 'TextStrikeThrough') {
+        ComponentName = 'TextStrikethrough';
+    }
+    if (ComponentName === 'UnLink') {
+        ComponentName = 'Unlink';
+    }
+
+    return ComponentName;
+};
```

Comment:
I feel like this could be simplified...

We could use some regex for removing the beginning or end of the string and use a replacements map to make the code less repetitive.

```suggestion
export const getComponentName = (i) => {
    let id = path
        .basename(i, '.svg') 
        .replace(/^(S2_Icon_|S_)/, '')  
        .replace(/(_20_N|_22x20_N|_18_N@2x)$/, ''); 

    if (i.startsWith('Ad')) {
        id = i.replace(/^Ad/, '') + 'Advert';
    }

    const replacements = {
        'UnLink': 'Unlink',
        'TextStrikeThrough': 'TextStrikethrough',
        'github': 'GitHub'
    };

    return Case.pascal(replacements[id] || id);
};
```